clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -238,15 +238,24 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
           instanceMetadata[(GoogleServerGroup.View.LOAD_BALANCING_POLICY)] = objectMapper.writeValueAsString(loadBalancingPolicy)
           backendToAdd = GCEUtil.backendFromLoadBalancingPolicy(loadBalancingPolicy)
         } else if (sourcePolicyJson) {
-          // We don't have to update the metadata here, since we are reading these properties directly from it.
-          backendToAdd = GCEUtil.backendFromLoadBalancingPolicy(objectMapper.readValue(sourcePolicyJson, GoogleHttpLoadBalancingPolicy))
+          GoogleHttpLoadBalancingPolicy newPolicy = objectMapper.readValue(sourcePolicyJson, GoogleHttpLoadBalancingPolicy)
+          Map sourcePolicyMap = objectMapper.readValue(sourcePolicyJson, Map)
```

Comment:
Could you just mark `listeningPort` as deprecated on the `GoogleHttpLoadBalancingPolicy` model, and just set it to null when necessary?

Reply:
SGTM